### DISCUSSION TITLE
想嘗試自動儲存png，但是圖片出不來

### DISCUSSION THREAD

**Discussion by bonny123865** [1101-GS3073-C/Python-Community-GS3073-C#52] [Q&A] [answered]
問題：
將一串程式輸入後，成果有出來，也有存成png檔案
但是打開後，喔買尬，卻只有空白的圖案
想請問一下該怎麼解決，還是我少加了什麼進去呢？

### **以下是輸入的程式**
![question](https://user-images.githubusercontent.com/90883526/136770737-b158bb23-547b-43fe-816b-fbbf97ea5d3d.png)

### **更以下是我能看到的圖檔**
![ohno](https://user-images.githubusercontent.com/90883526/136770747-4a9b3945-5d6f-4c85-8b54-ad4326159f3d.png)

**Comment by jacky0115** [comment, marked as answer]
plt.show()會重置目前的圖片(可以想成是畫完這個圖片物件之後可能會需要畫下一個，所以畫完之後立即創建一個空白的圖像物件等待下一次的plot)，所以導致後面使用的plt.savefig()會儲存到這張空白圖片
有兩種解決方法
1. plt.savefig()在plt.show()之前調用
2. 在畫圖前先獲取當前圖像物件
```
fig=plt.gcf()    #Get Current Figure
plt.show()
fig.savefig('squares_plot.png', dpi=100)
```

**Reply by bonny123865** [reply]
謝謝你，他成功顯示出來了！
